stack.yaml

```diff
@@ -131,3 +142,11 @@ ghc-options:
   cardano-sl:            -Werror
   cardano-sl-tools:      -Werror
   cardano-sl-lwallet:    -Werror
+  cardano-sl-explorer:   -Werror
+
+# Added for explorer
+flags:
+  cardano-sl:
+    with-explorer: True
```

Comment:
> I was talking about using stack build ... as a way to build the project. 

It doesn't matter `scripts/build/cardano-sl.sh` is just a wrapper over `stack build`.

> And that's because using stack build ... we implicitly include explorer since it's in packages.

It depends on which target you pass to `stack build`. If you use `stack build cardano-sl` it won't built explorer.

> The issue is building both projects without setting the flag to true.
> How are we going to build explorer with that flag set to False?

Once again, you should never build explorer without this flag. And I described it above. If one passes `--no-explorer` to build script, but wants to build explorer, it should be an error printed to stderr.